File: app/src/main/java/cmsr/ipsacademy/net/Activities/LoginActivity.java
package cmsr.ipsacademy.net.Activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import cmsr.ipsacademy.net.R;
import cmsr.ipsacademy.net.Util.SharedPreference;
import cmsr.ipsacademy.net.api.controller;
import cmsr.ipsacademy.net.api.responsemodel;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity
{
    private EditText editTextComputerCode, editTextPassword;
    private ImageView loginButton;
    private SharedPreference sharedPreference;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        sharedPreference = new SharedPreference(this);

        editTextComputerCode=(EditText)findViewById(R.id.edit_text_computer_code_login);
        editTextPassword=(EditText)findViewById(R.id.edit_text_password_login);
        loginButton=(ImageView)findViewById(R.id.login_button_image_submit);

        checkuserexistence();

        loginButton.setOnClickListener(view -> processlogin());

    }

    void processlogin()
    {
        String computer_code=editTextComputerCode.getText().toString().trim();
        String password=editTextPassword.getText().toString().trim();

        Call<responsemodel> call= controller
                .getInstance()
                .getapi()
                .verifyuser(computer_code,password);

        call.enqueue(new Callback<responsemodel>() {
            @Override
            public void onResponse(Call<responsemodel> call, Response<responsemodel> response) {
                responsemodel obj=response.body();
                String output=obj.getMessage();
                String is_student = obj.getIs_student();
                if(output.equals("failed"))
                {
                    editTextComputerCode.setText("");
                    editTextPassword.setText("");
                    Toast.makeText(getApplicationContext(), "Invalid username or password", Toast.LENGTH_SHORT).show();
                }
                if(output.equals("exist") && is_student.equals("yes"))
                {
                    Toast.makeText(getApplicationContext(), "Login Sucessfull", Toast.LENGTH_SHORT).show();
                    try {
                        sharedPreference.save("computer_code", editTextComputerCode.getText().toString().trim());
                    }catch (Exception e){
                        Log.e("e", e.toString());
                    }
                    startActivity(new Intent(getApplicationContext(),UserActivity.class));
                    finish();
                }
                if (output.equals("exist") && is_student.equals("no")){
                    Toast.makeText(getApplicationContext(), "Login Sucessfull", Toast.LENGTH_SHORT).show();
                    try {
                        sharedPreference.save("computer_code", editTextComputerCode.getText().toString().trim());
                    }catch (Exception e){
                        Log.e("e", e.toString());
                    }
                    startActivity(new Intent(getApplicationContext(),MainActivity.class));
                    finish();
                }
            }

            @Override
            public void onFailure(Call<responsemodel> call, Throwable t) {
                Log.i("failedLogin", t.toString());
            }
        });

    }

    void checkuserexistence()
    {
        try {
            if(sharedPreference.getValueString("computer_code")!=null)
                startActivity(new Intent(getApplicationContext(),UserActivity.class));
            else {
                Toast.makeText(getApplicationContext(), "Please Login", Toast.LENGTH_SHORT).show();
            }
        }catch (Exception e){
            Log.e("e", e.toString());
        }

    }
}

File: app/src/main/java/cmsr/ipsacademy/net/activities/LoginActivity.kt
package cmsr.ipsacademy.net.activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.KeyEvent
import android.view.View
import cmsr.ipsacademy.net.R
import cmsr.ipsacademy.net.api.apiset
import cmsr.ipsacademy.net.api.controller
import cmsr.ipsacademy.net.helpers.SharedPreferencesHelper

import android.widget.EditText
import android.widget.ImageView
import android.widget.Toast
import androidx.lifecycle.lifecycleScope
import cmsr.ipsacademy.net.activities.models.LoginModel
import cmsr.ipsacademy.net.helpers.AppConstants
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import okhttp3.Dispatcher
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.create

class LoginActivity : AppCompatActivity() {

    private var editTextComputerCode: EditText? = null
    private var editTextPassword: EditText? = null
    private var loginButton: ImageView? = null
    private var sharedPreference: SharedPreferencesHelper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        sharedPreference = SharedPreferencesHelper(this)

        editTextComputerCode = findViewById(R.id.edit_text_computer_code_login)
        editTextPassword = findViewById(R.id.edit_text_password_login)
        loginButton = findViewById(R.id.login_button_image_submit)

        checkUserExistence()

        editTextPassword?.setOnKeyListener(View.OnKeyListener { v, keyCode, event ->
            if (keyCode == KeyEvent.KEYCODE_ENTER && event.action == KeyEvent.ACTION_UP) {
                processLogin()
                return@OnKeyListener true
            }
            false
        })

        loginButton!!.setOnClickListener(View.OnClickListener {
            processLogin()
        })

    }

    private fun processLogin() {
        val computer_code = editTextComputerCode!!.text.toString().trim()
        val password = editTextPassword!!.text.toString().trim()

        lifecycleScope.launch(Dispatchers.IO) {
            val res = controller.getInstance().create(apiset::class.java)
                .verifyuser(computer_code, password).execute()

            if (res.body()?.student_info?.isEmpty() == true) {
                editTextComputerCode!!.setText("")
                editTextPassword!!.setText("")
                withContext(Dispatchers.Main) {
                    Toast.makeText(applicationContext, getString(R.string.login_error_message), Toast.LENGTH_SHORT).show()
                }
            }
            if (res.body()?.student_info?.isEmpty() == false && res.body()!!.student_info[0].is_student == "1") {
                withContext(Dispatchers.Main) {
                    Toast.makeText(applicationContext, getString(R.string.login_sucess), Toast.LENGTH_SHORT).show()
                }
                sharedPreference?.save(AppConstants.computer_code, computer_code)
                sharedPreference?.save(
                    AppConstants.user_role,
                    getString(R.string.role_student)
                )
                startActivity(Intent(applicationContext, MainActivity::class.java))
            }
            if (res.body()?.student_info?.isEmpty() == false && res.body()!!.student_info[0].is_student == "0") {

                withContext(Dispatchers.Main) {
                    Toast.makeText(applicationContext, getString(R.string.login_sucess), Toast.LENGTH_SHORT).show()
                }

                sharedPreference?.save(AppConstants.computer_code, computer_code)

                if (sharedPreference?.getValueString(AppConstants.computer_code)
                        .equals("2083")
                ) {
                    sharedPreference?.save(
                        AppConstants.user_role,
                        getString(R.string.role_hod)
                    )

                    sharedPreference?.getValueString(AppConstants.computer_code)
                        ?.let { Log.e("ROLESAvED", it) }
                    sharedPreference?.getValueString(AppConstants.user_role)
                        ?.let { Log.e("role", it) }
                } else if (sharedPreference?.getValueString(AppConstants.computer_code)
                        .equals("1") || sharedPreference?.getValueString(AppConstants.computer_code)
                        .equals("2") || sharedPreference?.getValueString(AppConstants.computer_code)
                        .equals("3")
                ) {
                    sharedPreference?.save(
                        AppConstants.user_role,
                        getString(R.string.role_principal)
                    )

                    sharedPreference?.getValueString(AppConstants.computer_code)
                        ?.let { Log.e("ROLESAvED", it) }
                    sharedPreference?.getValueString(AppConstants.user_role)
                        ?.let { Log.e("role", it) }
                } else {
                    sharedPreference?.save(
                        AppConstants.user_role,
                        getString(R.string.role_teacher)
                    )
                    sharedPreference?.getValueString(AppConstants.user_role)
                        ?.let { Log.i("role", it) }

                }

                startActivity(Intent(applicationContext, MainActivity::class.java))

            }
        }

    }

    private fun checkUserExistence() {
        if (sharedPreference?.getValueString(AppConstants.computer_code) != null) {
            if (sharedPreference?.getValueString(AppConstants.user_role)
                    .equals(getString(R.string.role_student))
            ) {
                startActivity(Intent(applicationContext, MainActivity::class.java))
            } else if (sharedPreference?.getValueString(AppConstants.user_role)
                    .equals(getString(R.string.role_hod))
            ) {
                startActivity(Intent(applicationContext, UserActivity::class.java))
            } else if (sharedPreference?.getValueString(AppConstants.user_role)
                    .equals(getString(R.string.role_teacher))
            ) {
                startActivity(Intent(applicationContext, MainActivity::class.java))
            } else if (sharedPreference?.getValueString(AppConstants.user_role)
                    .equals(getString(R.string.role_principal))
            ) {
                startActivity(Intent(applicationContext, UserActivity::class.java))
            }

        } else {
            Toast.makeText(applicationContext, getString(R.string.please_login), Toast.LENGTH_SHORT)
                .show()
        }
    }

}